src/Smart.Design.Console/Models/Validations/GenerateOptionsValidator.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using FluentValidation;
+using Smart.Design.Console.CommandOptions;
+using Smart.Design.RazorView;
+
+namespace Smart.Design.Console.Models.Validations
+{
+    public class GenerateOptionsValidator : AbstractValidator<GenerateOptions>
+    {
+        public GenerateOptionsValidator()
+        {
+            RuleFor(options => options.DesignElementName)
+                .Must(elementName =>
+                {
+                    var customTagHelpersCollection = new CustomTagHelperCollections();
+                    return customTagHelpersCollection.Names.Contains(elementName);
+                })
+                .WithMessage(errorMessage => "Can't find the name indicated among the existing components");
```

Comment:
In some place, you're talking about "design elements", and here about "components". Is there a difference ?